In [7]:
# Leer el dataset "Dataset2" de la carpeta Datasets/Crudos
import pandas as pd
data1 = pd.read_csv('/Users/carolinasv/Documents/VS_Code/pulsera_inteligente/Datasets/Crudos/Dataset2.csv')
pd.reset_option('display.max_rows')
data1

itching  skin_rash  nodal_skin_eruptions  continuous_sneezing  \
0           1          1                     1                    0   
1           0          1                     1                    0   
2           1          0                     1                    0   
3           1          1                     0                    0   
4           1          1                     1                    0   
...       ...        ...                   ...                  ...   
4915        0          0                     0                    0   
4916        0          1                     0                    0   
4917        0          0                     0                    0   
4918        0          1                     0                    0   
4919        0          1                     0                    0   

      shivering  chills  joint_pain  stomach_pain  acidity  ulcers_on_tongue  \
0             0       0           0             0        0                 0   
1             0       0           0             0        0                 0   
2             0       0           0             0        0                 0   
3             0       0           0             0        0                 0   
4             0       0           0             0        0                 0   
...         ...     ...         ...           ...      ...               ...   
4915          0       0           0             0        0                 0   
4916          0       0           0             0        0                 0   
4917          0       0           0             0        0                 0   
4918          0       0           1             0        0                 0   
4919          0       0           0             0        0                 0   

      ...  scurring  skin_peeling  silver_like_dusting  small_dents_in_nails  \
0     ...         0             0                    0                     0   
1     ...         0             0                    0                     0   
2     ...         0             0                    0                     0   
3     ...         0             0                    0                     0   
4     ...         0             0                    0                     0   
...   ...       ...           ...                  ...                   ...   
4915  ...         0             0                    0                     0   
4916  ...         1             0                    0                     0   
4917  ...         0             0                    0                     0   
4918  ...         0             1                    1                     1   
4919  ...         0             0                    0                     0   

      inflammatory_nails  blister  red_sore_around_nose  yellow_crust_ooze  \
0                      0        0                     0                  0   
1                      0        0                     0                  0   
2                      0        0                     0                  0   
3                      0        0                     0                  0   
4                      0        0                     0                  0   
...                  ...      ...                   ...                ...   
4915                   0        0                     0                  0   
4916                   0        0                     0                  0   
4917                   0        0                     0                  0   
4918                   1        0                     0                  0   
4919                   0        1                     1                  1   

                                    prognosis  Unnamed: 133  
0                            Fungal infection           NaN  
1                            Fungal infection           NaN  
2                            Fungal infection           NaN  
3                            Fungal infection 

In [4]:
# Mostrar todas las columnas del dataset
# Convertir el objeto data1.columns a una lista
lista_columnas = list(data1.columns)

# Imprimir la lista de columnas
print(lista_columnas)

['itching', 'skin_rash', 'nodal_skin_eruptions', 'continuous_sneezing', 'shivering', 'chills', 'joint_pain', 'stomach_pain', 'acidity', 'ulcers_on_tongue', 'muscle_wasting', 'vomiting', 'burning_micturition', 'spotting_ urination', 'fatigue', 'weight_gain', 'anxiety', 'cold_hands_and_feets', 'mood_swings', 'weight_loss', 'restlessness', 'lethargy', 'patches_in_throat', 'irregular_sugar_level', 'cough', 'high_fever', 'sunken_eyes', 'breathlessness', 'sweating', 'dehydration', 'indigestion', 'headache', 'yellowish_skin', 'dark_urine', 'nausea', 'loss_of_appetite', 'pain_behind_the_eyes', 'back_pain', 'constipation', 'abdominal_pain', 'diarrhoea', 'mild_fever', 'yellow_urine', 'yellowing_of_eyes', 'acute_liver_failure', 'fluid_overload', 'swelling_of_stomach', 'swelled_lymph_nodes', 'malaise', 'blurred_and_distorted_vision', 'phlegm', 'throat_irritation', 'redness_of_eyes', 'sinus_pressure', 'runny_nose', 'congestion', 'chest_pain', 'weakness_in_limbs', 'fast_heart_rate', 'pain_during_bow

In [36]:
# Cambiar la opción 'display.max_rows' a None
pd.set_option('display.max_rows', None)

# Convertir el objeto data1.columns a una lista
lista_columnas = list(data1.columns)

# Convertir la lista de columnas en un DataFrame
df_columnas = pd.DataFrame(lista_columnas, columns=['Enfermedad'])

# Mostrar el DataFrame
df_columnas

Enfermedad
0                           itching
1                         skin_rash
2              nodal_skin_eruptions
3               continuous_sneezing
4                         shivering
5                            chills
6                        joint_pain
7                      stomach_pain
8                           acidity
9                  ulcers_on_tongue
10                   muscle_wasting
11                         vomiting
12              burning_micturition
13              spotting_ urination
14                          fatigue
15                      weight_gain
16                          anxiety
17             cold_hands_and_feets
18                      mood_swings
19                      weight_loss
20                     restlessness
21                         lethargy
22                patches_in_throat
23            irregular_sugar_level
24                            cough
25                       high_fever
26                      sunken_eyes
27                   breathlessness
28                         sweating
29                      dehydration
30                      indigestion
31                         headache
32                   yellowish_skin
33                       dark_urine
34                           nausea
35                 loss_of_appetite
36             pain_behind_the_eyes
37                        back_pain
38                     constipation
39                   abdominal_pain
40                        diarrhoea
41                       mild_fever
42                     yellow_urine
43                yellowing_of_eyes
44              acute_liver_failure
45                   fluid_overload
46              swelling_of_stomach
47              swelled_lymph_nodes
48                          malaise
49     blurred_and_distorted_vision
50                           phlegm
51                throat_irritation
52                  redness_of_eyes
53                   sinus_pressure
54                       runny_nose
55                       congestion
56                       chest_pain
57                weakness_in_limbs
58                  fast_heart_rate
59      pain_during_bowel_movements
60              pain_in_anal_region
61                     bloody_stool
62               irritation_in_anus
63                        neck_pain
64                        dizziness
65                           cramps
66                         bruising
67                          obesity
68                     swollen_legs
69            swollen_blood_vessels
70              puffy_face_and_eyes
71                 enlarged_thyroid
72                    brittle_nails
73              swollen_extremeties
74                 excessive_hunger
75           extra_marital_contacts
76         drying_and_tingling_lips
77                   slurred_speech
78                        knee_pain
79                   hip_joint_pain
80                  muscle_weakness
81                       stiff_neck
82                  swelling_joints
83               movement_stiffness
84               spinning_movements
85                  loss_of_balance
86                     unsteadiness
87        weakness_of_one_body_side
88                    loss_of_smell
89               bladder_discomfort
90              foul_smell_of urine
91         continuous_feel_of_urine
92                 passage_of_gases
93                 internal_itching
94              toxic_look_(typhos)
95                       depression
96                     irritability
97                      muscle_pain
98                altered_sensorium
99              red_spots_over_body
100                      belly_pain
101           abnormal_menstruation
102             dischromic _patches
103              watering_from_eyes
104              increased_appetite
105                        polyuria
106                  family_history
107                   mucoid_sputum
108                    rusty_sputum
109           lack_of_concentration
110             visual_distur

# Traducción de síntomas
- 0 picazón (X)
- 1 erupción cutánea
- 2 erupciones_de_piel_nodal
- 3 estornudos continuos (X)
- 4 temblando (X)
- 5 escalofríos (X)
- 6 dolor_articular (X)
- 7 dolor de estómago (X)
- 8 acidez (X)
- 9 úlceras_en_la_lengua
- 10 pérdida de músculo
- 11 vómitos (X)
- 12 micción_ardor
- 13 manchado_ orinar
- 14 fatiga (X)
- 15 aumento de peso
- 16 ansiedad
- 17 manos_y_pies_frios (X)
- 18 cambios de humor (X)
- 19 pérdida de peso
- 20 inquietud
- 21 letargo
- 22 parches_en_la_garganta (X)
- 23 nivel_de_azúcar_irregular
- 24 tos (X)
- 25 fiebre_alta (X)
- 26 ojos hundidos
- 27 falta de aire (X)
- 28 sudando (X)
- 29 deshidratación
- 30 indigestión (X)
- 31 dolor de cabeza (X)
- 32 piel_amarillenta
- 33 orina_oscura
- 34 náuseas (X)
- 35 pérdida_de_apetito
- 36 dolor_detrás_de_los_ojos (X)
- 37 dolor de espalda (X)
- 38 estreñimiento (X)
- 39 dolor_abdominal (X)
- 40 diarrea (X)
- 41 fiebre_leve (X)
- 42 orina_amarilla
- 43 ojos_amarillentos
- 44 insuficiencia_hepática_aguda
- 45 sobrecarga_fluido
- 46 hinchazón_de_estómago
- 47 ganglios linfáticos inflamados (X)
- 48 malestar (X)
- 49 visión_borrosa_y_distorsionada (X)
- 50 flema (X)
- 51 irritación_de_garganta (X)
- 52 enrojecimiento_de_ojos (X)
- 53 presión_sinusal (X)
- 54 nariz que moquea (X)
- 55 congestión (X)
- 56 dolor_pecho (X)
- 57 debilidad_en_extremidades (X)
- 58 ritmo_corazón_rápido (X)
- 59 dolor_durante_los_movimientos_intestinales (X)
- 60 dolor_en_región_anal
- 61 taburete_sangriento
- 62 irritación_en_ano
- 63 dolor de cuello
- 64 mareos (X)
- 65 calambres
- 66 hematomas
- 67 obesidad
- 68 piernas_hinchadas
- 69 vasos sanguíneos inflamados
- 70 cara_y_ojos_hinchados
- 71 tiroides agrandada
- 72 uñas quebradizas
- 73 extremidades hinchadas
- 74 hambre excesiva
- 75 contactos_extra_maritales
- 76 labios_secado_y_hormigueo
- 77 dificultad para hablar (X)
- 78 dolor de rodilla
- 79 dolor_articular_cadera
- 80 debilidad_muscular
- 81 cuello rígido
- 82 articulaciones hinchadas
- 83 rigidez_movimiento (X)
- 84 movimientos_giratorios (X)
- 85 pérdida_de_equilibrio (X)
- 86 inestabilidad (X)
- 87 debilidad_de_un_lado_del_cuerpo
- 88 pérdida_del_olfato (X)
- 89 malestar_vejiga
- 90 mal olor a orina
- 91 sensación_continua_de_orina
- 92 paso_de_gases
- 93 picazón interna
- 94 mirada_tóxica_(tifos)
- 95 depresión
- 96 irritabilidad (X)
- 97 dolor_muscular (X)
- 98 sensorio_alterado 
- 99 manchas_rojas_sobre_el_cuerpo
- 100 dolor de vientre
- 101 menstruación_anormal
- 102 parches discrómicos
- 103 lagrimeo_de_ojos (X)
- 104 aumento_apetito
- 105 poliuria
- 106 historia_familiar
- 107 esputo_mucoide
- 108 esputo_oxidado
- 109 falta_de_concentración (X)
- 110 perturbaciones_visuales (X)
- 111 recibiendo_transfusión_de_sangre
- 112 recibiendo_inyecciones_no estériles
- 113 coma
- 114 sangrado_estomacal (X)
- 115 distensión_del_abdomen
- 116 historia_del_consumo_de_alcohol
- 117 sobrecarga_fluido.1
- 118 sangre_en_esputo
- 119 venas_prominentes_en_la_calf
- 120 palpitaciones (X)
- 121 caminar_doloroso
- 122 granos_llenos_de_pus
- 123 puntos negros
- 124 corriendo
- 125 pelado de piel
- 126 polvo_como_plata
- 127 pequeñas_dentaciones_en_uñas
- 128 uñas_inflamatorias
- 129 ampolla
- 130 dolor_rojo_alrededor_de_la_nariz (X)
- 131 amarillo_crust_ooze

# 2do Filtro - Traducción de síntomas
- 0 picazón (X) --> itching
- 4 temblando (X) --> shivering
- 5 escalofríos (X) --> chills
- 7 dolor de estómago (X) --> stomach_pain
- 11 vómitos (X) --> vomiting
- 14 fatiga (X) --> fatigue
- 17 manos_y_pies_frios (X) --> cold_hands_and_feets
- 24 tos (X) --> cough
- 25 fiebre_alta (X) --> high_fever
- 27 falta de aire (X) --> breathlessness
- 31 dolor de cabeza (X) --> headache
- 34 náuseas (X) --> nausea
- 39 dolor_abdominal (X) --> abdominal_pain
- 41 fiebre_leve (X) --> mild_fever
- 48 malestar (X) --> malaise
- 51 irritación_de_garganta (X) --> throat_irritation
- 52 enrojecimiento_de_ojos (X) --> redness_of_eyes
- 55 congestión (X) --> congestion
- 58 ritmo_corazón_rápido (X) --> fast_heart_rate
- 64 mareos (X) --> dizziness
- 83 rigidez_movimiento (X) --> movement_stiffness
- 96 irritabilidad (X) --> irritability
- 103 lagrimeo_de_ojos (X) --> watering_from_eyes
- 109 falta_de_concentración (X) --> lack_of_concentration
- 120 palpitaciones (X) --> palpitations

# Lista de síntomas 
## Temperatura
### Síntomas asociados
- Fiebre
- Hipotermia
## Información que da el sensor
- Temperatura del ambiente
- Temperatura del objeto

## Pulsioxímetro
### Síntomas asociados
- Taquicardia
- Bradicardia
- Hipoxia
- Hiperoxemia o hiperoxia
## Información que da el sensor
- Frecuencia cardiaca 
- Saturación de oxígeno

## Infrarojo Co2 (Enfermedad hipercapnia)
### Síntomas asociados
#### Concentración alta de CO2
- Dolor de cabeza
- Mareos
- Dificultad para respirar
- Somnolencia

#### Concentración baja de CO2
- Confusión mental
- Naúseas
- Mareos
- Malestar estomacal
- Entumecimiento y hormigueo
- Pérdida del conocimiento
- Dificultad para respirar

#### Concentración alta de compuestos orgánicos
- Tos
- Dolor de garganta
- Congestión nasal
- Dificultad para respirar
- Irritación en los ojos (enrojecimiento, picazón, ardor y lagrimeo)
- Dolor de cabeza
- Mareos
- Náuseas
- Aturdimiento
- Fatiga
- Debilidad general
- Falta de energía
- Confusión
- Dificultad para concentrarse
- Irritabilidad

#### Concentración baja de compuestos orgánicos
- Irritación (Ojos, nariz, garganta)
- Picor
- Quemazón
- Dolor de cabeza
- Mareo
- Fatiga
- Nauseas
- Perdida de coordinación

## Información que da el sensor
- Concentración de CO2
- Concentración y de compuestos orgánicos volátiles

In [9]:
# Crear un dataset con columas especificadas de data1
lista_filtrada = ["itching","shivering","chills","stomach_pain","vomiting", "fatigue", "cold_hands_and_feets", 
                    "cough", "high_fever", "breathlessness", "headache", "nausea", "abdominal_pain", 
                    "mild_fever", "malaise","throat_irritation","redness_of_eyes","congestion","fast_heart_rate",
                    "dizziness","movement_stiffness","irritability","watering_from_eyes","lack_of_concentration",
                    "palpitations", "prognosis"]
data2 = data1[lista_filtrada]
pd.reset_option('display.max_rows')
data2

itching  shivering  chills  stomach_pain  vomiting  fatigue  \
0           1          0       0             0         0        0   
1           0          0       0             0         0        0   
2           1          0       0             0         0        0   
3           1          0       0             0         0        0   
4           1          0       0             0         0        0   
...       ...        ...     ...           ...       ...      ...   
4915        0          0       0             0         1        0   
4916        0          0       0             0         0        0   
4917        0          0       0             0         0        0   
4918        0          0       0             0         0        0   
4919        0          0       0             0         0        0   

      cold_hands_and_feets  cough  high_fever  breathlessness  ...  \
0                        0      0           0               0  ...   
1                        0      0           0               0  ...   
2                        0      0           0               0  ...   
3                        0      0           0               0  ...   
4                        0      0           0               0  ...   
...                    ...    ...         ...             ...  ...   
4915                     0      0           0               0  ...   
4916                     0      0           0               0  ...   
4917                     0      0           0               0  ...   
4918                     0      0           0               0  ...   
4919                     0      0           1               0  ...   

      redness_of_eyes  congestion  fast_heart_rate  dizziness  \
0                   0           0                0          0   
1                   0           0                0          0   
2                   0           0                0          0   
3                   0           0                0          0   
4                   0           0                0          0   
...               ...         ...              ...        ...   
4915                0           0                0          0   
4916                0           0                0          0   
4917                0           0                0          0   
4918                0           0                0          0   
4919                0           0                0          0   

      movement_stiffness  irritability  watering_from_eyes  \
0                      0             0                   0   
1                      0             0                   0   
2                      0             0                   0   
3                      0             0                   0   
4                      0             0                   0   
...                  ...           ...                 ...   
4915                   0             0                   0   
4916                   0             0                   0   
4917                   0             0                   0   
4918                   0             0                   0   
4919                   0             0                   0   

      lack_of_concentration  palpitations  \
0                         0             0   
1                         0             0   
2                         0             0   
3                         0             0   
4                         0             0   
...                     ...           ...   
4915                      0             0   
4916                      0             0   
4917                      0             0   
4918                      0             0   
4919                      0             0   

                                    prognosis  
0                            Fungal infection  
1                            Fungal infection  
2                            Fungal infection  
3                            Fungal infection  
4                            

In [10]:
# Crear un dataset con columas especificadas de data1
lista_filtrada2 = ["itching","shivering","chills","stomach_pain","vomiting", "fatigue", "cold_hands_and_feets", 
                    "cough", "high_fever", "breathlessness", "headache", "nausea", "abdominal_pain", 
                    "mild_fever", "malaise","throat_irritation","redness_of_eyes","congestion","fast_heart_rate",
                    "dizziness","movement_stiffness","irritability","watering_from_eyes","lack_of_concentration",
                    "palpitations"]
data3 = data1[lista_filtrada2]
pd.reset_option('display.max_rows')
print(data3.head())

   itching  shivering  chills  stomach_pain  vomiting  fatigue  \
0        1          0       0             0         0        0   
1        0          0       0             0         0        0   
2        1          0       0             0         0        0   
3        1          0       0             0         0        0   
4        1          0       0             0         0        0   

   cold_hands_and_feets  cough  high_fever  breathlessness  ...  \
0                     0      0           0               0  ...   
1                     0      0           0               0  ...   
2                     0      0           0               0  ...   
3                     0      0           0               0  ...   
4                     0      0           0               0  ...   

   throat_irritation  redness_of_eyes  congestion  fast_heart_rate  dizziness  \
0                  0                0           0                0          0   
1                  0                0 

In [11]:
# Mostrar los valores únicos de la columna 'prognosis'
data2['prognosis'].unique()

array(['Fungal infection', 'Allergy', 'GERD', 'Chronic cholestasis',
       'Drug Reaction', 'Peptic ulcer diseae', 'AIDS', 'Diabetes ',
       'Gastroenteritis', 'Bronchial Asthma', 'Hypertension ', 'Migraine',
       'Cervical spondylosis', 'Paralysis (brain hemorrhage)', 'Jaundice',
       'Malaria', 'Chicken pox', 'Dengue', 'Typhoid', 'hepatitis A',
       'Hepatitis B', 'Hepatitis C', 'Hepatitis D', 'Hepatitis E',
       'Alcoholic hepatitis', 'Tuberculosis', 'Common Cold', 'Pneumonia',
       'Dimorphic hemmorhoids(piles)', 'Heart attack', 'Varicose veins',
       'Hypothyroidism', 'Hyperthyroidism', 'Hypoglycemia',
       'Osteoarthristis', 'Arthritis',
       '(vertigo) Paroymsal  Positional Vertigo', 'Acne',
       'Urinary tract infection', 'Psoriasis', 'Impetigo'], dtype=object)

In [17]:
# Contar los valores únicos de la columna 'prognosis'
print(data2['prognosis'].nunique())

41


## Traducción
['Infección por hongos', 'Alergia', 'ERGE', 'Colestasis crónica',
 'Reacción a medicamentos', 'Úlcera péptica', 'SIDA', 'Diabetes',
 'Gastroenteritis', 'Asma Bronquial', 'Hipertensión', 'Migraña',
 'Espondilosis cervical', 'Parálisis (hemorragia cerebral)', 'Ictericia',
 'Malaria', 'Varicela', 'Dengue', 'Tifoidea', 'hepatitis A',
 'Hepatitis B', 'Hepatitis C', 'Hepatitis D', 'Hepatitis E',
 'Hepatitis alcohólica', 'Tuberculosis', 'Resfriado común', 'Neumonía',
 'Hemorroides (almorranas) dimórficas', 'Ataque al corazón', 'Varices',
 'Hipotiroidismo', 'Hipertiroidismo', 'Hipoglucemia',
 'Osteoartritis', 'Artritis',
 '(vértigo) Vértigo posicional paroymsal', 'Acné',
 'Infección del tracto urinario', 'Psoriasis', 'Impetigo']

In [14]:
# Eliminar los registros que tienen todos los valores iguales a 0 en las columnas de síntomas excepto 'prognosis'
data4 = data2[(data2.iloc[:, :-1] != 0).any(axis=1)]
pd.reset_option('display.max_rows')
data4

itching  shivering  chills  stomach_pain  vomiting  fatigue  \
0           1          0       0             0         0        0   
2           1          0       0             0         0        0   
3           1          0       0             0         0        0   
4           1          0       0             0         0        0   
6           1          0       0             0         0        0   
...       ...        ...     ...           ...       ...      ...   
4911        0          0       0             0         0        1   
4912        0          0       0             0         1        1   
4914        0          0       0             0         0        0   
4915        0          0       0             0         1        0   
4919        0          0       0             0         0        0   

      cold_hands_and_feets  cough  high_fever  breathlessness  ...  \
0                        0      0           0               0  ...   
2                        0      0           0               0  ...   
3                        0      0           0               0  ...   
4                        0      0           0               0  ...   
6                        0      0           0               0  ...   
...                    ...    ...         ...             ...  ...   
4911                     0      0           0               0  ...   
4912                     0      0           0               0  ...   
4914                     0      0           0               0  ...   
4915                     0      0           0               0  ...   
4919                     0      0           1               0  ...   

      redness_of_eyes  congestion  fast_heart_rate  dizziness  \
0                   0           0                0          0   
2                   0           0                0          0   
3                   0           0                0          0   
4                   0           0                0          0   
6                   0           0                0          0   
...               ...         ...              ...        ...   
4911                0           0                1          0   
4912                0           0                0          0   
4914                0           0                0          0   
4915                0           0                0          0   
4919                0           0                0          0   

      movement_stiffness  irritability  watering_from_eyes  \
0                      0             0                   0   
2                      0             0                   0   
3                      0             0                   0   
4                      0             0                   0   
6                      0             0                   0   
...                  ...           ...                 ...   
4911                   0             1                   0   
4912                   0             1                   0   
4914                   1             0                   0   
4915                   0             0                   0   
4919                   0             0                   0   

      lack_of_concentration  palpitations  \
0                         0             0   
2                         0             0   
3                         0             0   
4                         0             0   
6                         0             0   
...                     ...           ...   
4911                      0             0   
4912                      0             1   
4914                      0             0   
4915                      0             0   
4919                      0             0   

                                    prognosis  
0                            Fungal infection  
2                            Fungal infection  
3                            Fungal infection  
4                            Fungal infection  
6                            

In [15]:
# Mostrar los valores únicos de la columna 'prognosis'
data4['prognosis'].unique()

array(['Fungal infection', 'Allergy', 'GERD', 'Chronic cholestasis',
       'Drug Reaction', 'Peptic ulcer diseae', 'AIDS', 'Diabetes ',
       'Gastroenteritis', 'Bronchial Asthma', 'Hypertension ', 'Migraine',
       'Cervical spondylosis', 'Paralysis (brain hemorrhage)', 'Jaundice',
       'Malaria', 'Chicken pox', 'Dengue', 'Typhoid', 'hepatitis A',
       'Hepatitis B', 'Hepatitis C', 'Hepatitis D', 'Hepatitis E',
       'Alcoholic hepatitis', 'Tuberculosis', 'Common Cold', 'Pneumonia',
       'Heart attack', 'Varicose veins', 'Hypothyroidism',
       'Hyperthyroidism', 'Hypoglycemia', 'Arthritis',
       '(vertigo) Paroymsal  Positional Vertigo', 'Impetigo'],
      dtype=object)

In [27]:
# Contar los valores únicos de la columna 'prognosis'
print(data4['prognosis'].nunique())

36


In [28]:
# Contabilizar el número de registros por cada valor único de la columna 'prognosis'
data4['prognosis'].value_counts()

prognosis
Typhoid                                    120
Dengue                                     120
Hyperthyroidism                            120
Hypothyroidism                             120
Heart attack                               120
Pneumonia                                  120
Common Cold                                120
Tuberculosis                               120
Alcoholic hepatitis                        120
Hepatitis E                                120
Hepatitis D                                120
Hepatitis C                                120
Hepatitis B                                120
hepatitis A                                120
Allergy                                    120
Chicken pox                                120
(vertigo) Paroymsal  Positional Vertigo    120
Bronchial Asthma                           120
GERD                                       120
Chronic cholestasis                        120
Drug Reaction                              120
Pep

In [26]:
# Contabilizar el número de registros con dos valores igual a 1 en una de las columnas de síntomas
print(data4.iloc[:, :-1].sum(axis=1).value_counts())

1    1002
2     846
4     660
3     624
5     330
7     306
6     276
8     120
9      84
Name: count, dtype: int64


In [29]:
# Eliminar los registros que solo tengan el valor 1 en uno o dos campos de síntomas
data5 = data4[(data4.iloc[:, :-1].sum(axis=1) > 2)]
data5

itching  shivering  chills  stomach_pain  vomiting  fatigue  \
10          0          1       1             0         0        0   
11          0          1       1             0         0        0   
15          0          1       1             0         0        0   
19          0          1       1             0         0        0   
20          0          0       0             1         1        0   
...       ...        ...     ...           ...       ...      ...   
4906        0          0       1             0         0        1   
4910        0          0       0             0         0        1   
4911        0          0       0             0         0        1   
4912        0          0       0             0         1        1   
4915        0          0       0             0         1        0   

      cold_hands_and_feets  cough  high_fever  breathlessness  ...  \
10                       0      0           0               0  ...   
11                       0      0           0               0  ...   
15                       0      0           0               0  ...   
19                       0      0           0               0  ...   
20                       0      1           0               0  ...   
...                    ...    ...         ...             ...  ...   
4906                     0      1           1               1  ...   
4910                     1      0           0               0  ...   
4911                     0      0           0               0  ...   
4912                     0      0           0               0  ...   
4915                     0      0           0               0  ...   

      redness_of_eyes  congestion  fast_heart_rate  dizziness  \
10                  0           0                0          0   
11                  0           0                0          0   
15                  0           0                0          0   
19                  0           0                0          0   
20                  0           0                0          0   
...               ...         ...              ...        ...   
4906                0           0                1          0   
4910                0           0                0          1   
4911                0           0                1          0   
4912                0           0                0          0   
4915                0           0                0          0   

      movement_stiffness  irritability  watering_from_eyes  \
10                     0             0                   1   
11                     0             0                   1   
15                     0             0                   1   
19                     0             0                   1   
20                     0             0                   0   
...                  ...           ...                 ...   
4906                   0             0                   0   
4910                   0             1                   0   
4911                   0             1                   0   
4912                   0             1                   0   
4915                   0             0                   0   

      lack_of_concentration  palpitations  \
10                        0             0   
11                        0             0   
15                        0             0   
19                        0             0   
20                        0             0   
...                     ...           ...   
4906                      0             0   
4910                      0             0   
4911                      0             0   
4912                      0             1   
4915                      0             0   

                                    prognosis  
10                                    Allergy  
11                                    Allergy  
15                                    Allergy  
19                                    Allergy  
20                           

In [30]:
# Contabilizar el número de registros de data5
print(data5.shape[0])

2400


In [31]:
# Mostrar los valores únicos de la columna 'prognosis'
data5['prognosis'].unique()

array(['Allergy', 'GERD', 'Chronic cholestasis', 'Bronchial Asthma',
       'Hypertension ', 'Jaundice', 'Malaria', 'Chicken pox', 'Dengue',
       'Typhoid', 'hepatitis A', 'Hepatitis B', 'Hepatitis D',
       'Hepatitis E', 'Tuberculosis', 'Common Cold', 'Pneumonia',
       'Hypothyroidism', 'Hyperthyroidism', 'Hypoglycemia',
       '(vertigo) Paroymsal  Positional Vertigo'], dtype=object)

In [32]:
# Contar los valores únicos de la columna 'prognosis'
print(data5['prognosis'].nunique())

21


array(['Alergia', 'ERGE', 'Colestasis crónica', 'Asma bronquial',
 'Hipertensión', 'Ictericia', 'Malaria', 'Varicela', 'Dengue',
 'Tifoidea', 'hepatitis A', 'Hepatitis B', 'Hepatitis D',
 'Hepatitis E', 'Tuberculosis', 'Resfriado común', 'Neumonía',
 'Hipotiroidismo', 'Hipertiroidismo', 'Hipoglucemia',
 '(vértigo) Vértigo posicional paroymsal'], dtype=object)

In [36]:
# Reemplazar los valores de la columna 'prognosis' por valores de otro arreglo
data6 = data5.copy()
data6['prognosis'] = data6['prognosis'].replace(
    ['Allergy', 'GERD', 'Chronic cholestasis', 'Bronchial Asthma',
       'Hypertension ', 'Jaundice', 'Malaria', 'Chicken pox', 'Dengue',
       'Typhoid', 'hepatitis A', 'Hepatitis B', 'Hepatitis D',
       'Hepatitis E', 'Tuberculosis', 'Common Cold', 'Pneumonia',
       'Hypothyroidism', 'Hyperthyroidism', 'Hypoglycemia',
       '(vertigo) Paroymsal  Positional Vertigo'],
    ['Alergia', 'ERGE', 'Colestasis crónica', 'Asma bronquial','Hipertensión', 'Ictericia', 'Malaria', 
     'Varicela', 'Dengue', 'Tifoidea', 'hepatitis A', 'Hepatitis B', 'Hepatitis D', 'Hepatitis E', 
     'Tuberculosis', 'Resfriado común', 'Neumonía', 'Hipotiroidismo', 'Hipertiroidismo', 'Hipoglucemia',
     '(Vértigo) Vértigo posicional paroymsal'])

data6

itching  shivering  chills  stomach_pain  vomiting  fatigue  \
10          0          1       1             0         0        0   
11          0          1       1             0         0        0   
15          0          1       1             0         0        0   
19          0          1       1             0         0        0   
20          0          0       0             1         1        0   
...       ...        ...     ...           ...       ...      ...   
4906        0          0       1             0         0        1   
4910        0          0       0             0         0        1   
4911        0          0       0             0         0        1   
4912        0          0       0             0         1        1   
4915        0          0       0             0         1        0   

      cold_hands_and_feets  cough  high_fever  breathlessness  ...  \
10                       0      0           0               0  ...   
11                       0      0           0               0  ...   
15                       0      0           0               0  ...   
19                       0      0           0               0  ...   
20                       0      1           0               0  ...   
...                    ...    ...         ...             ...  ...   
4906                     0      1           1               1  ...   
4910                     1      0           0               0  ...   
4911                     0      0           0               0  ...   
4912                     0      0           0               0  ...   
4915                     0      0           0               0  ...   

      redness_of_eyes  congestion  fast_heart_rate  dizziness  \
10                  0           0                0          0   
11                  0           0                0          0   
15                  0           0                0          0   
19                  0           0                0          0   
20                  0           0                0          0   
...               ...         ...              ...        ...   
4906                0           0                1          0   
4910                0           0                0          1   
4911                0           0                1          0   
4912                0           0                0          0   
4915                0           0                0          0   

      movement_stiffness  irritability  watering_from_eyes  \
10                     0             0                   1   
11                     0             0                   1   
15                     0             0                   1   
19                     0             0                   1   
20                     0             0                   0   
...                  ...           ...                 ...   
4906                   0             0                   0   
4910                   0             1                   0   
4911                   0             1                   0   
4912                   0             1                   0   
4915                   0             0                   0   

      lack_of_concentration  palpitations  \
10                        0             0   
11                        0             0   
15                        0             0   
19                        0             0   
20                        0             0   
...                     ...           ...   
4906                      0             0   
4910                      0             0   
4911                      0             0   
4912                      0             1   
4915                      0             0   

                                   prognosis  
10                                   Alergia  
11                                   Alergia  
15                                   Alergia  
19                                   Alergia  
20                                

In [37]:
# Contabilizar el número de registros de data5
print(data6.shape[0])

2400


In [38]:
# Exportar data6 a un archivo CSV
data6.to_csv('/Users/carolinasv/Documents/VS_Code/pulsera_inteligente/Datasets/Procesados/Dataset2_procesado.csv', index=False)